### PR TITLE
Consistent overwrite

### PR DESCRIPTION
**Pull request by LukasNickel**
- Remove all -f aliases
- Check for conflict on setup
- rename infile/outfile to input_path/output_path
- Use ToolConfigurationError everywhere

I guess that would be an API change since we remove a flag from some tools?

### CODE REVIEW

File: ctapipe/tools/dump_triggers.py [hunk 1]
```diff
@@ -103,10 +101,15 @@ def add_event_to_table(self, event):
         )
 
     def setup(self):
-        """ setup function, called before `start()` """
+        """setup function, called before `start()`"""
 
-        if self.infile == "":
-            raise ToolConfigurationError("No 'infile' parameter was specified. ")
+        if self.input_path == "":
+            raise ToolConfigurationError("No 'input_path' parameter was specified. ")
```
Comment (nbiederbeck) [inline ctapipe/tools/dump_triggers.py:107]: this can be set via `allow_none=False` in L25

**Review by maxnoe — COMMENTED** [review]
Now this could be a function ...

**Review by nbiederbeck — APPROVED** [review]

File: ctapipe/tools/dump_triggers.py [hunk 2]
```diff
@@ -103,10 +101,15 @@ def add_event_to_table(self, event):
         )
 
     def setup(self):
-        """ setup function, called before `start()` """
+        """setup function, called before `start()`"""
 
-        if self.infile == "":
-            raise ToolConfigurationError("No 'infile' parameter was specified. ")
+        if self.output_path.exists():
```
Comment (maxnoe) [inline ctapipe/tools/dump_triggers.py:106]: Use the new method also here?

File: ctapipe/core/tool.py [hunk 3]
```diff
@@ -324,6 +325,26 @@ def add_component(self, component_instance):
         self._registered_components.append(component_instance)
         return component_instance
 
+    def check_output(self, output_paths):
```
Comment (maxnoe) [inline ctapipe/core/tool.py:328]: `*output_path`

File: ctapipe/tools/dump_triggers.py [hunk 4]
```diff
@@ -23,31 +22,28 @@ class DumpTriggersTool(Tool):
...
-    overwrite = Bool(False, help="overwrite existing output file").tag(config=True)
-
     # =============================================
     # map low-level options to high-level command-line options
     # =============================================
 
     aliases = Dict(
-        {"infile": "DumpTriggersTool.infile", "outfile": "DumpTriggersTool.outfile"}
+        {
+            "input_path": "DumpTriggersTool.input_path",
```
Comment (maxnoe) [inline ctapipe/tools/dump_triggers.py:41]: Please change to `--input` and `--output` to be consistent with the other tools.

**Review by maxnoe — DISMISSED** [review]

File: docs/changes/2213.api.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Remove `-f` flag as alias for `--overwrite` and fail early if output exists, but overwrite is not set
```
Comment (nbiederbeck) [inline docs/changes/2213.api.rst:1]: These to be double backticks, sorry.  Docs are failing bc of this.

**Review by nbiederbeck — CHANGES_REQUESTED** [review]

**Review by nbiederbeck — APPROVED** [review]

**Comment by nbiederbeck** [comment]
With the newest changes I think we would benefit from a test, checking that overwriting works explicitly.

**Review by kosack — DISMISSED** [review]

File: ctapipe/core/tool.py [hunk 6]
```diff
@@ -324,6 +329,32 @@ def add_component(self, component_instance):
...
+        Parameters
+        ----------
+        output_paths: Path
+            One or more output path to check.
+
+        """
+        for output in output_paths:
+            if output is not None and output.exists():
+                if self.overwrite:
+                    self.log.warning(f"Overwriting {output}")
```
Comment (kosack) [inline ctapipe/core/tool.py:352]: if you want Codacy to stop complaining, use 
```python
self.log.warning("Overwriting %s", output)
```
Not really critical though!

**Review by maxnoe — APPROVED** [review]

**Review by nbiederbeck — APPROVED** [review]

File: ctapipe/tools/train_energy_regressor.py [hunk 7]
```diff
@@ -98,18 +97,7 @@ def setup(self):
         self.cross_validate = CrossValidator(
             parent=self, model_component=self.regressor
         )
-        self.rng = np.random.default_rng(self.random_seed)
```
Comment (maxnoe) [inline ctapipe/tools/train_energy_regressor.py:101]: This line should have stayed, review missed it and no tests for `n_events`

**Pull request merged**